plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -88,7 +88,7 @@ public function get_formatted_log( $level, $message, $context = array() ) {
 			unset( $context['tags'] );
 		}
 
-		if ( class_exists( '\WC_Tracks' ) ) {
+		if ( class_exists( '\WC_Tracks' ) && function_exists( 'wp_get_current_user' ) ) {
 			$user         = wp_get_current_user();
 			$blog_details = \WC_Tracks::get_blog_details( $user->ID );
```

Comment:
Interestingly, `WC_Tracks::get_blog details` uses the user ID to get the site's language -- which I think is technically incorrect considering the name of the function. It would be more accurate to get the site language setting with something like `get_bloginfo('language')` rather than that of the current user.

Potentially we change the WC_Tracks class to use the site language and won't need to query user info here 🤔

Reply:
We could also reduce our dependency on `WC_Tracks` entirely and determine `blog_id` and `store_id` values independently.

```
$blog_id  = class_exists( 'Jetpack_Options' ) ? Jetpack_Options::get_option( 'id' ) : null
$store_id' = get_option( \WC_Install::STORE_ID_OPTION, null );
```

The dependency on [Jetpack_Options](https://github.com/Automattic/jetpack-options) could perhaps also be removed and just make a straight call to WP `get_option` 🤔

Reply:
I'll make a separate issue for that since the scope of those changes have a bigger blast radius than the ones I've made

Reply:
Issue created: https://github.com/woocommerce/woocommerce/issues/51310